### PR TITLE
feat(ecocredit): add tx origin duplication check to `CreateBatch` and `MintBatchCredits`

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Ref: #1084 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/create_batch_test.go [hunk 1]
```diff
@@ -121,6 +121,88 @@ func TestCreateBatch_ProjectNotFound(t *testing.T) {
...
+		EndDate:   &end,
+		OriginTx: &core.OriginTx{
+			Typ: "Ethereum",
+			Id:  "210985091248",
+		},
+	})
+	assert.NilError(t, err)
+}
+
+func TestCreateBatch_WithOriginTx_InValid(t *testing.T) {
```
Comment (ryanchristo) [inline x/ecocredit/server/core/create_batch_test.go:162]:
```suggestion
func TestCreateBatch_WithOriginTx_Invalid(t *testing.T) {
```

File: x/ecocredit/server/core/create_batch_test.go [hunk 2]
```diff
@@ -121,6 +121,88 @@ func TestCreateBatch_ProjectNotFound(t *testing.T) {
...
+	t.Parallel()
+	s := setupBase(t)
+	batchTestSetup(t, s.ctx, s.stateStore, s.addr)
+	_, _, addr2 := testdata.KeyTestPubAddr()
+
+	blockTime, err := time.Parse("2006-01-02", "2049-01-30")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/create_batch_test.go:168]: Same as above. Would be nice to use `types.ParseDate` here.

File: x/ecocredit/server/core/create_batch_test.go [hunk 3]
```diff
@@ -121,6 +121,88 @@ func TestCreateBatch_ProjectNotFound(t *testing.T) {
...
+			Id:  "210985091248",
+		},
+	}
+
+	_, err = s.k.CreateBatch(s.ctx, batch)
+	assert.NilError(t, err)
+
+	// create credit batch with same tx origin id
+	_, err = s.k.CreateBatch(s.ctx, batch)
+	assert.ErrorContains(t, err, "credits already issued")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/create_batch_test.go:203]:
```suggestion
	assert.ErrorContains(t, err, "credits already issued with tx id")
```

File: x/ecocredit/server/core/mint_batch_credits_test.go [hunk 4]
```diff
@@ -93,6 +101,37 @@ func TestMintBatchCredits_NotFound(t *testing.T) {
...
+			Id:  "210985091248",
+		},
+		Note: "bridged credits",
+	}
+
+	_, err := s.k.MintBatchCredits(ctx, &msg)
+	assert.NilError(t, err)
+
+	_, err = s.k.MintBatchCredits(ctx, &msg)
+	assert.ErrorContains(t, err, "credits already issued")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/mint_batch_credits_test.go:132]:
```suggestion
	assert.ErrorContains(t, err, "credits already issued with tx id")
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me. Just a few nits. Nice work.

**Review by technicallyty — APPROVED** [review]
💯

**Pull request merged**